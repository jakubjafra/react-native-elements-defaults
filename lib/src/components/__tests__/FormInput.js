const testHelperUtils = require('../../testHelperUtils');

const utils = require('../../utils');
const Overwrite = utils.Overwrite;

const name = 'FormInput',
    __oldElement__ = require('react-native-elements/src/form/FormInput'),
    __element__ = require('../FormInput'),
    noTheme = new Overwrite(
        name,
        Overwrite.TYPE.CLASS_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {}
    ),
    basicTheme = new Overwrite(
        name,
        Overwrite.TYPE.CLASS_COMPONENT_ATTRIBUTES,
        Overwrite.SCOPE.DEFAULTS,
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );

describe(name, () => {
    testHelperUtils.defaultTests(name, __oldElement__, __element__, noTheme, basicTheme);
});
