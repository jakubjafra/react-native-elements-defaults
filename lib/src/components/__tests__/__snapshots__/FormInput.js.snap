// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput with basic theme applied FormInput renders correctly 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "red",
      "borderBottomColor": "#bdc6cf",
      "borderBottomWidth": 1,
      "marginLeft": 20,
      "marginRight": 20,
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
    }
  }
>
  <TextInput
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#86939e",
          "minHeight": 36,
          "width": 750,
        },
        Object {
          "fontSize": 17.5,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`FormInput with basic theme applied OldFormInput renders correctly 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "borderBottomColor": "#bdc6cf",
      "borderBottomWidth": 1,
      "marginLeft": 20,
      "marginRight": 20,
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
    }
  }
>
  <TextInput
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#86939e",
          "minHeight": 36,
          "width": 750,
        },
        Object {
          "fontSize": 17.5,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`FormInput with no theme applied FormInput renders correctly 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "borderBottomColor": "#bdc6cf",
      "borderBottomWidth": 1,
      "marginLeft": 20,
      "marginRight": 20,
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
    }
  }
>
  <TextInput
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#86939e",
          "minHeight": 36,
          "width": 750,
        },
        Object {
          "fontSize": 17.5,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`FormInput with no theme applied OldFormInput and FormInput returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`FormInput with no theme applied OldFormInput renders correctly 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "borderBottomColor": "#bdc6cf",
      "borderBottomWidth": 1,
      "marginLeft": 20,
      "marginRight": 20,
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
    }
  }
>
  <TextInput
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#86939e",
          "minHeight": 36,
          "width": 750,
        },
        Object {
          "fontSize": 17.5,
        },
        undefined,
      ]
    }
  />
</View>
`;
